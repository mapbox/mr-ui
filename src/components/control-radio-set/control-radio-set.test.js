import React from 'react';
import { shallow } from 'enzyme';
import { testCases } from './control-radio-set-test-cases';

describe('ControlRadioSet', () => {
  let testCase;
  let wrapper;

  describe(testCases.basic.description, () => {
    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('onChange gets called with input id and value', () => {
      const mockEvent = {
        target: {
          value: 'foo'
        }
      };
      wrapper.find('input').first().props().onChange(mockEvent);
      wrapper.update();
      expect(wrapper).toMatchSnapshot();
      expect(testCase.props.onChange).toHaveBeenCalledTimes(1);
      expect(testCase.props.onChange).toHaveBeenCalledWith('foo', 'testinput');
    });
  });

  describe(testCases.disabled.description, () => {
    beforeEach(() => {
      testCase = testCases.disabled;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('extra props works', () => {
      const props = wrapper.find('input').last().props();
      expect(props).toHaveProperty('disabled', true);
    });
  });

  describe(testCases.allOptions.description, () => {
    beforeEach(() => {
      testCase = testCases.allOptions;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
