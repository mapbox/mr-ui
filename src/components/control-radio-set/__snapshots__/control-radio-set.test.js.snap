// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlRadioSet all options renders 1`] = `
<ControlWrapper
  id="testinput"
  themeControlWrapper="bg-red-light"
  validationError="oh no!"
>
  <div
    aria-invalid={true}
    aria-required={false}
    data-test="testinput-input"
    id="testinput"
    role="radiogroup"
  >
    <label
      className="radio-container w-full"
      key="Option value 1"
    >
      <input
        autoFocus={true}
        checked={true}
        name="testinput"
        onChange={[Function]}
        type="radio"
        value="Option value 1"
      />
      <div
        className="radio--white radio--active-red mr6 radio"
      />
      <span>
        Option 1
      </span>
    </label>
    <label
      className="radio-container w-full"
      key="Option value 2"
    >
      <input
        autoFocus={false}
        checked={false}
        name="testinput"
        onChange={[Function]}
        type="radio"
        value="Option value 2"
      />
      <div
        className="radio--white radio--active-red mr6 radio"
      />
      <span>
        Option 2
      </span>
    </label>
    <label
      className="radio-container w-full"
      key="Option value 3"
    >
      <input
        autoFocus={false}
        checked={false}
        name="testinput"
        onChange={[Function]}
        type="radio"
        value="Option value 3"
      />
      <div
        className="radio--white radio--active-red mr6 radio"
      />
      <span>
        Option 3
      </span>
    </label>
  </div>
</ControlWrapper>
`;

exports[`ControlRadioSet basic onChange gets called with input id and value 1`] = `
<ControlWrapper
  id="testinput"
>
  <div
    aria-required={true}
    data-test="testinput-input"
    id="testinput"
    role="radiogroup"
  >
    <label
      className="radio-container txt-s block mb6 flex"
      key="Option value 1"
    >
      <input
        autoFocus={false}
        checked={false}
        name="testinput"
        onChange={[Function]}
        type="radio"
        value="Option value 1"
      />
      <div
        className="mr6 radio--blue radio--s-label inline-block  radio"
      />
      Option 1
    </label>
    <label
      className="radio-container txt-s block mb6 flex"
      key="Option value 2"
    >
      <input
        autoFocus={false}
        checked={false}
        name="testinput"
        onChange={[Function]}
        type="radio"
        value="Option value 2"
      />
      <div
        className="mr6 radio--blue radio--s-label inline-block  radio"
      />
      Option 2
    </label>
  </div>
</ControlWrapper>
`;

exports[`ControlRadioSet basic renders 1`] = `
<ControlWrapper
  id="testinput"
>
  <div
    aria-required={true}
    data-test="testinput-input"
    id="testinput"
    role="radiogroup"
  >
    <label
      className="radio-container txt-s block mb6 flex"
      key="Option value 1"
    >
      <input
        autoFocus={false}
        checked={false}
        name="testinput"
        onChange={[Function]}
        type="radio"
        value="Option value 1"
      />
      <div
        className="mr6 radio--blue radio--s-label inline-block  radio"
      />
      Option 1
    </label>
    <label
      className="radio-container txt-s block mb6 flex"
      key="Option value 2"
    >
      <input
        autoFocus={false}
        checked={false}
        name="testinput"
        onChange={[Function]}
        type="radio"
        value="Option value 2"
      />
      <div
        className="mr6 radio--blue radio--s-label inline-block  radio"
      />
      Option 2
    </label>
  </div>
</ControlWrapper>
`;

exports[`ControlRadioSet disabled renders 1`] = `
<ControlWrapper
  id="testinput"
>
  <div
    aria-required={true}
    data-test="testinput-input"
    id="testinput"
    role="radiogroup"
  >
    <label
      className="radio-container txt-s block mb6 flex"
      key="Option value 1"
    >
      <input
        autoFocus={false}
        checked={false}
        name="testinput"
        onChange={[Function]}
        type="radio"
        value="Option value 1"
      />
      <div
        className="mr6 radio--blue radio--s-label inline-block  radio"
      />
      Option 1
    </label>
    <label
      className="radio-container txt-s block mb6 flex"
      key="Option value 2"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={true}
        name="testinput"
        onChange={[Function]}
        type="radio"
        value="Option value 2"
      />
      <div
        className="mr6 radio--blue radio--s-label inline-block  radio"
      />
      Option 2
    </label>
  </div>
</ControlWrapper>
`;
